dotcom-rendering/src/web/components/Picture.tsx

```diff
@@ -131,6 +131,27 @@ const getDesiredWidthForBreakpoint = (
 		return breakpoint;
 	}
 
+	if (isMainMedia && format.display === ArticleDisplay.Immersive) {
+		// Immersive main media images are often reported to have poor quality so we
+		// override the default values here and force larger images to be used
+		switch (breakpoint) {
+			case breakpoints.mobile:
+				return 480;
+			case breakpoints.mobileLandscape:
+				return 660;
+			case breakpoints.phablet:
+				return 740;
+			case breakpoints.tablet:
+				return 980;
+			case breakpoints.desktop:
+				return 1140;
+			case breakpoints.leftCol:
+				return 1300;
+			case breakpoints.wide:
+				return 1900;
+		}
```

Comment:
It wasn’t clear to me that what we were doing here is getting the larger breakpoints.

Instead of semi-magic numbers, could we make it more explicit like:

```suggestion
		switch (breakpoint) {
			case breakpoints.mobile:
				return breakpoints.mobileLandscape;
			case breakpoints.mobileLandscape:
				return breakpoints.phablet;
			case breakpoints.phablet:
				return breakpoints.tablet;
			case breakpoints.tablet:
				return breakpoints.desktop;
			case breakpoints.desktop:
				return breakpoints.leftCol;
			case breakpoints.leftCol:
				return breakpoints.wide;
			case breakpoints.wide:
				return 1900;
		}
```

Or alternatively do an actual review of the perfect values and stray away from breakpoints

```suggestion
		switch (breakpoint) {
			case breakpoints.mobile:
				return 500;
			case breakpoints.mobileLandscape:
				return 700;
			case breakpoints.phablet:
				return 800;
			case breakpoints.tablet:
				return 1000;
			case breakpoints.desktop:
				return 1200;
			case breakpoints.leftCol:
				return 1300;
			case breakpoints.wide:
				return 1900;
		}
```

Reply:
These numbers are actually very specific for a reason: 

<img width="369" alt="image" src="https://user-images.githubusercontent.com/9575458/159456198-b2640f97-255c-4ffc-9ca2-38e96ca7aa8b.png">

We have to stay faithful to the sources provided to us by frontend, otherwise we'll select images of an ever higher resolution unintentionally

Reply:
I think magic numbers can be a problem but I think in this case these values are extremely unlikely to change so they are already like constants - you see them in the code a lot - so maybe having them written inline isn't so bad? It's nice to be able to read them straight away without another level of abstraction in-between.

But I also see the case for using a named var in this case where it's not clear what's happening in this logic (it's choosing the next level _up_).

But I think we should address this in a separate PR. I don't want to introduce an inconsistent pattern in this function where some code returns numbers and other code returns named vars.

Reply:
> But I also see the case for using a named var in this case where it's not clear what's happening in this logic (it's choosing the next level _up_).

I think this is what tripped me the most. But really these values are not picking larger breakpoints, they’re matching frontend logic, and the fact that these numbers are the same as breakpoints is _somewhat accidental_.

https://github.com/guardian/frontend/blob/6a144f59373929664e7b84148c7699df93dbc63d/common/app/layout/ContentWidths.scala#L104-L116